src/http_server/api/v1/health.c

```diff
@@ -24,11 +24,33 @@
 #include <fluent-bit/flb_pack.h>
 #include <fluent-bit/flb_sds.h>
 #include <fluent-bit/flb_macros.h>
+#include <fluent-bit/flb_utils.h>
 #include <fluent-bit/flb_http_server.h>
 #include <msgpack.h>
 
 #include "health.h"
 
+/*
+ * in/out records sample at a certain timestamp.
+ */
+struct flb_hs_throughput_sample {
+    uint64_t in_records;
+    uint64_t out_records;
+    uint64_t timestamp_seconds;
+    struct mk_list _head;
+};
+
+struct {
```

Comment:
Is there absolutely no other way to do this than using a global anonymous structure?

Reply:
Can you clarify what is your suggestion/request here?

Reply:
I was asking you to move the definitions of both struct types to the header file. Considering the preexisting code and how it handles `metrics_counter` I don't know if I'd change the fact that you're using a global instance of the structure as I don't think turning it into a dynamically allocated yet shared structure would be beneficial and introducing the TLS as is would even be detrimental to maintainability.

So, to clarify, move the type definitions to the header file. leave the global variable definition in the code file.